### ISSUE TITLE
Kernel dies at the "Extracting topics from clusters using representation models" stage

### ISSUE THREAD

**Issue by pjacobetty** [MaartenGr/BERTopic#1836]
I used to run BERTopic without a problem on my job's HPC on GPUs but now it seems like it doesn't move beyond this stage:
2024-02-23 12:24:56,663 - BERTopic - Representation - Extracting topics from clusters using representation models.
It kills the ipython kernel and it doesn't provide any other info, even though I have verbose=True.

Can you please help me?
Thanks!

**Comment by pjacobetty** [comment]
Hi Maarten, Thanks for the prompt reply. Your tool is incredibly useful and I really appreciate your work.
I'm closing this because I eventually managed to identify the problem: the jupyter binary path was not pointing to the venv despite being active. 
Thanks again! Great work.

**Comment by mayingjie0809** [comment]
Hi Maarten, thanks for your wonderful work! I'm working on a large dataset containing 700K+ documents and I face the similar problem.

I'm running BERTopic model step by step on Google Colab with T4 GPU (high RAM mode, 51GB system RAM, 15GB GPU RAM, 235.7GB disk). 
`
import os
import re
import pandas as pd
import numpy as np
from tqdm import tqdm

import gensim.corpora as corpora
from gensim.models.coherencemodel import CoherenceModel

from bertopic import BERTopic
from sentence_transformers import SentenceTransformer
from cuml.cluster import HDBSCAN
from cuml.manifold import UMAP
from bertopic.representation import KeyBERTInspired, PartOfSpeech
from bertopic.vectorizers import ClassTfidfTransformer
from keyphrase_vectorizers import KeyphraseTfidfVectorizer
import nltk

nltk.download('stopwords')

import spacy
print(spacy.require_gpu()) # The answer is TRUE
nlp = spacy.load("en_core_web_sm")

import cupy
import cupyx
cupy.cuda.runtime.getDeviceCount()

# Here I load the pre-trained embeddings
embedding_model = SentenceTransformer('mpi-inno-comp/paecter')
embeddings = np.load("/content/drive/MyDrive/embeddings_01.npy")

umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine', random_state=42)
hdbscan_model = HDBSCAN(min_cluster_size=10, min_samples=10, gen_min_span_tree=True, metric='euclidean', cluster_selection_method='eom', prediction_data=True)
vectorizer_model = KeyphraseTfidfVectorizer(stop_words="english")
ctfidf_model = ClassTfidfTransformer(bm25_weighting=True, reduce_frequent_words=True)

keybert_model = KeyBERTInspired()
pos_model = PartOfSpeech(model=nlp)
representation_model = [keybert_model, pos_model]

topic_model = BERTopic(
    embedding_model=embedding_model,
    umap_model=umap_model,
    hdbscan_model=hdbscan_model,
    vectorizer_model=vectorizer_model,
    ctfidf_model=ctfidf_model,
    representation_model=representation_model,
    nr_topics='auto',
    top_n_words=20,
    calculate_probabilities=True,
    verbose=True
)

topics, probs = topic_model.fit_transform(texts, embeddings)
topic_model.save("/content/drive/MyDrive/", serialization="safetensors", save_ctfidf=True, save_embedding_model=embedding_model)`

After running, it spent about 10+mins finishing dimensionality reduction and clustering with cuML. However, when processing following models, it seems kind of tricky.
`2025-04-28 05:59:31,143 - BERTopic - Dimensionality - Fitting the dimensionality reduction algorithm
2025-04-28 06:09:47,254 - BERTopic - Dimensionality - Completed ✓
2025-04-28 06:09:47,270 - BERTopic - Cluster - Start clustering the reduced embeddings
2025-04-28 06:11:55,967 - BERTopic - Cluster - Completed ✓
**2025-04-28 06:11:55,971 - BERTopic - Representation - Extracting topics using c-TF-IDF for topic reduction.**`
After 9+ hours, it was still stuck at this step. Therefore, I'm wondering if you have any good suggestions on this? Sincere thanks again

**Comment by MaartenGr** [comment]
@mayingjie0809 My guess is the `KeyphraseTfidfVectorizer`, that tends to take a lot of compute to run. Have you tried it without it?

**Comment by MaartenGr** [comment]
@mayingjie0809 Yes, you can just remove the `vectorizer_model` parameter.

**Comment by mayingjie0809** [comment]
It does works for me after removing the **vectorizer_model** parameter!!! Great thanks for your support!
